packages/xo-web/src/xo-app/vm/tab-stats.js

```diff
@@ -1,182 +1,163 @@
       clearTimeout(this.timeout)
-
       this.setState(
         {
-          granularity,
-          selectStatsLoading: true,
+          stats,
+          selectStatsLoading: false,
         },
-        this.loop
+        () => {
+          this.timeout = setTimeout(this.loop, stats.interval * 1000)
+        }
       )
+    })
+  }
+  loop = ::this.loop
+
+  componentWillMount() {
+    this.loop()
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeout)
+  }
+
+  componentWillReceiveProps(props) {
+    const vmCur = this.props.vm
+    const vmNext = props.vm
+
+    if (vmCur.power_state !== 'Running' && vmNext.power_state === 'Running') {
+      this.loop(vmNext)
+    } else if (
+      vmCur.power_state === 'Running' &&
+      vmNext.power_state !== 'Running'
+    ) {
+      this.setState({
+        stats: undefined,
+      })
     }
-    handleSelectStats = ::this.handleSelectStats
+  }
 
-    render() {
-      const { intl } = this.props
-      const {
+  handleSelectStats(granularity) {
+    clearTimeout(this.timeout)
+
+    this.setState(
+      {
         granularity,
-        selectStatsLoading,
-        stats,
-        useCombinedValues,
-      } = this.state
+        selectStatsLoading: true,
+      },
+      this.loop
+    )
+  }
+  handleSelectStats = ::this.handleSelectStats
 
-      return !stats ? (
-        <p>No stats.</p>
-      ) : (
-        <Container>
-          <Row>
-            <Col mediumSize={6}>
-              <div className='form-group'>
-                <Tooltip content={_('useStackedValuesOnStats')}>
-                  <Toggle
-                    value={useCombinedValues}
-                    onChange={this.linkState('useCombinedValues')}
-                  />
-                </Tooltip>
-              </div>
-              {selectStatsLoading && (
-                <div className='text-xs-right'>
-                  <Icon icon='loading' size={2} />
-                </div>
-              )}
-            </Col>
-            <Col mediumSize={6}>
-              <div className='btn-tab'>
-                <select
-                  className='form-control'
-                  onChange={this.handleSelectStats}
-                  defaultValue={granularity}
-                >
-                  <option value='seconds'>
-                    {intl.formatMessage(messages.statLastTenMinutes)}
-                  </option>
-                  <option value='minutes'>
-                    {intl.formatMessage(messages.statLastTwoHours)}
-                  </option>
-                  <option value='hours'>
-                    {intl.formatMessage(messages.statLastWeek)}
-                  </option>
-                  <option value='days'>
-                    {intl.formatMessage(messages.statLastYear)}
-                  </option>
-                </select>
+  render() {
+    const {
+      granularity,
+      selectStatsLoading,
+      stats,
+      useCombinedValues,
+    } = this.state
+
+    return !stats ? (
```

Comment:
Make it an excplicit test please

Reply:
This change is not related to this PR. In this render i just replace the old select by the mutualized select.